sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -111,31 +102,80 @@ case class GpuGetArrayItem(child: Expression, ordinal: Expression, failOnError:
   override def nullable: Boolean = true
   override def dataType: DataType = child.dataType.asInstanceOf[ArrayType].elementType
 
-  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector =
-    throw new IllegalStateException("This is not supported yet")
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    val (array, indices) = (lhs.getBase, rhs.getBase)
+    val indicesCol = withResource(Scalar.fromInt(0)) { zeroS =>
+      withResource(indices.lessThan(zeroS)) { hasNegativeIndicesCV =>
+        val hasNegativeIndices = isAnyValidTrue(hasNegativeIndicesCV)
+        if (failOnError) {
+          // Check if any index is out of bound only when ansi mode is enabled.
+          // No exception should be raised if no valid entry (An entry is valid when both
+          // the array row and its index are not null), the same with what Spark does.
+          if(hasNegativeIndices && array.getNullCount != array.getRowCount) {
```

Comment:
nit: in general I think we want to have a space after the `if`. There are several places where this is not happening, and what is more it is inconsistent in the patch.

Reply:
updated